File: src/game/save/motto.ts
import { MOTTO_MAX_LENGTH } from './types'

/** Trim, collapse whitespace, and cap length. Empty becomes null. */
export function normalizeMotto(raw: string): string | null {
  const trimmed = raw.trim().replace(/\s+/g, ' ')
  if (!trimmed) return null
  return trimmed.slice(0, MOTTO_MAX_LENGTH)
}

export function isValidMotto(raw: string): boolean {
  // Empty is allowed (clears the motto); only reject when over the cap after trim.
  const trimmed = raw.trim().replace(/\s+/g, ' ')
  return trimmed.length <= MOTTO_MAX_LENGTH
}

File: packages/ik_rules/lib/src/save/motto.dart
import 'generated/save_models.dart';

final RegExp _whitespaceRun = RegExp(r'\s+');

/// Trim, collapse whitespace, and cap length. Empty becomes null.
String? normalizeMotto(String raw) {
  final trimmed = raw.trim().replaceAll(_whitespaceRun, ' ');
  if (trimmed.isEmpty) return null;
  return trimmed.length <= mottoMaxLength ? trimmed : trimmed.substring(0, mottoMaxLength);
}

bool isValidMotto(String raw) {
  // Empty is allowed (clears the motto); only reject when over the cap after trim.
  final trimmed = raw.trim().replaceAll(_whitespaceRun, ' ');
  return trimmed.length <= mottoMaxLength;
}
